graylog2-server/src/main/java/org/graylog/security/authservice/AuthServiceBackend.java

```diff
@@ -41,6 +41,19 @@
 
     default Optional<AuthenticationDetails> authenticateAndProvision(AuthServiceToken token,
             ProvisionerService provisionerService) {
+        return authenticateAndProvision(token, provisionerService, true);
+    }
+
+    /**
+     * Authenticate user and provision a local user record
+     * @param token
+     * @param provisionerService
+     * @param isExternal user is managed by external service (e.g. Okta) and cannot be modified locally
+     * @return authentication details; empty if a problem was encountered
+     */
+    default Optional<AuthenticationDetails> authenticateAndProvision(AuthServiceToken token,
+                                                                     ProvisionerService provisionerService,
+                                                                     boolean isExternal) {
```

Comment:
Do we still need this change? I don't see it being used with the current state of the implementation so I vote for reverting it to keep the interface clean.